lib/tasks/data/remove_transient_data.rb

```diff
@@ -1,20 +1,26 @@
 # frozen_string_literal: true
 
 class RemoveTransientData
-  RETENTION_PERIOD = 3.months.ago.to_date
+  RETENTION_PERIOD = 3.months
 
   # This model lets us operate on the sessions DB table using ActiveRecord's
   # methods within the scope of this service. This relies on the AR's
   # convention where a Session model maps to a sessions table.
   class Session < ApplicationRecord
   end
 
+  attr_reader :expiration_date
+
+  def initialize
+    @expiration_date = RETENTION_PERIOD.ago.to_date
+  end
+
   def call
     Rails.logger.info("#{self.class.name}: processing")
 
-    Spree::StateChange.where("created_at < ?", RETENTION_PERIOD).delete_all
-    Spree::LogEntry.where("created_at < ?", RETENTION_PERIOD).delete_all
-    Session.where("updated_at < ?", RETENTION_PERIOD).delete_all
+    Spree::StateChange.where("created_at < ?", @expiration_date).delete_all
```

Comment:
It would be great if we could use the attr_reader for `@expiration_date` in this class as well i.e. `expiration_date`.